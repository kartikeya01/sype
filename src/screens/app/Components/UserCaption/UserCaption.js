import React from 'react';
import {Subheading} from 'react-native-paper';
import {Text, StyleSheet} from 'react-native';
import PropTypes from 'prop-types';

export default class UserCaption extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <Subheading style={styles.SubheadingStyle}>
       
        <Text style={styles.captionStyles, styles.boldStyle}>{this.props.caption}</Text>
      </Subheading>
    );
  }
}
UserCaption.propTypes = {
  caption: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
};

const styles = StyleSheet.create({
  boldStyle: {
    fontWeight: 'bold',
    color: '#3700b3'
  },
  captionStyles: {  justifyContent: 'center', flexWrap: 'wrap'},
});
