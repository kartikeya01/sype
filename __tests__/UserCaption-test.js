import 'react-native';
import React from 'react';
import UserCaption from '@screens/app/Components/UserCaption/UserCaption';
import checkPropTypes from 'check-prop-types';
var assert = require('assert');
import {shallow} from 'enzyme';

import renderer from 'react-test-renderer';

test('user caption snapshot test', () => {
  const tree = renderer
    .create(<UserCaption username="test" caption="test" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
// test numbers, arrays, strings, booleans combinations of other data types
describe('testing UserCaption values, should not throw errors', () => {
  const stringCaptionTest = 'Hi World, Missed Me ?';
  const stringUsernameTest = 'e805';
  it('testing UserCaption Creation String', () => {
    const wrapper = shallow(
      <UserCaption username={stringUsernameTest} caption={stringCaptionTest} />,
    );
    const instance = wrapper.instance();
    const currentUsername = instance.state.username;
    const currentCaption = instance.state.caption;
    expect(currentUsername).toEqual(stringUsernameTest);
    expect(currentCaption).toEqual(stringCaptionTest);
  });
});
describe('testing UserCaption values, should throw errors', () => {
  it('Throws failed propType Error for not providing caption', () => {
    let result = checkPropTypes(
      UserCaption.propTypes,
      {username: 'bsefa'},
      'prop',
      UserCaption.username,
    );
    assert(
      result ===
        'Failed prop type: The prop `caption` is marked as required in `<<anonymous>>`, but its value is `undefined`.',
    );
  });
  it('Throws failed propType Error for not providing username', () => {
    let result = checkPropTypes(
      UserCaption.propTypes,
      {caption: 'corona is wack'},
      'prop',
      UserCaption.caption,
    );
    assert(
      result ===
        'Failed prop type: The prop `username` is marked as required in `<<anonymous>>`, but its value is `undefined`.',
    );
  });
  it('Throws failed propType Error for not providing caption', () => {
    let result = checkPropTypes(UserCaption.propTypes, {}, 'prop', UserCaption);
    assert(
      result.includes(
        'Failed prop type: The prop `caption` is marked as required',
      ),
    );
  });
  it('Throws failed propType Error for providing username value of true and not string', () => {
    let result = checkPropTypes(
      UserCaption.propTypes,
      {username: true, caption: 'corona is wack'},
      'prop',
      UserCaption.username,
    );
    assert(
      result ===
        'Failed prop type: Invalid prop `username` of type `boolean` supplied to `<<anonymous>>`, expected `string`.',
    );
  });
  it('Throws failed propType Error for providing username value of false and not string', () => {
    let result = checkPropTypes(
      UserCaption.propTypes,
      {username: false, caption: 'corona is wack'},
      'prop',
      UserCaption.username,
    );
    assert(
      result ===
        'Failed prop type: Invalid prop `username` of type `boolean` supplied to `<<anonymous>>`, expected `string`.',
    );
  });
  it('Throws failed propType Error for providing username value of undefined and not string', () => {
    let result = checkPropTypes(
      UserCaption.propTypes,
      {username: undefined, caption: 'corona is wack'},
      'prop',
      UserCaption.username,
    );
    assert(
      result ===
        'Failed prop type: The prop `username` is marked as required in `<<anonymous>>`, but its value is `undefined`.',
    );
  });
  it('Throws failed propType Error for providing username value of number and not string', () => {
    let result = checkPropTypes(
      UserCaption.propTypes,
      {username: 234, caption: 'corona is wack'},
      'prop',
      UserCaption.username,
    );
    assert(
      result ===
        'Failed prop type: Invalid prop `username` of type `number` supplied to `<<anonymous>>`, expected `string`.',
    );
  });
  it('Throws failed propType Error for providing username value of number and not string', () => {
    let result = checkPropTypes(
      UserCaption.propTypes,
      {username: 234.5, caption: 'corona is wack'},
      'prop',
      UserCaption.username,
    );
    assert(
      result ===
        'Failed prop type: Invalid prop `username` of type `number` supplied to `<<anonymous>>`, expected `string`.',
    );
  });
  it('Throws failed propType Error for providing caption value of true and not string', () => {
    let result = checkPropTypes(
      UserCaption.propTypes,
      {caption: true, username: 'emanasia'},
      'prop',
      UserCaption.username,
    );
    assert(
      result ===
        'Failed prop type: Invalid prop `caption` of type `boolean` supplied to `<<anonymous>>`, expected `string`.',
    );
  });
  it('Throws failed propType Error for providing caption value of false and not string', () => {
    let result = checkPropTypes(
      UserCaption.propTypes,
      {caption: false, username: 'emanasia'},
      'prop',
      UserCaption.username,
    );
    assert(
      result ===
        'Failed prop type: Invalid prop `caption` of type `boolean` supplied to `<<anonymous>>`, expected `string`.',
    );
  });
  it('Throws failed propType Error for providing caption value of undefined and not string', () => {
    let result = checkPropTypes(
      UserCaption.propTypes,
      {caption: undefined, username: 'emanasia'},
      'prop',
      UserCaption.username,
    );
    assert(
      result ===
        'Failed prop type: The prop `caption` is marked as required in `<<anonymous>>`, but its value is `undefined`.',
    );
  });
  it('Throws failed propType Error for providing caption value of number and not string', () => {
    let result = checkPropTypes(
      UserCaption.propTypes,
      {caption: 234, username: 'emanasia'},
      'prop',
      UserCaption.username,
    );
    assert(
      result ===
        'Failed prop type: Invalid prop `caption` of type `number` supplied to `<<anonymous>>`, expected `string`.',
    );
  });
  it('Throws failed propType Error for providing caption value of number and not string', () => {
    let result = checkPropTypes(
      UserCaption.propTypes,
      {caption: 234.5, username: 'emanasia'},
      'prop',
      UserCaption.username,
    );
    assert(
      result ===
        'Failed prop type: Invalid prop `caption` of type `number` supplied to `<<anonymous>>`, expected `string`.',
    );
  });
});

//TODO : test handleCommentClick function
