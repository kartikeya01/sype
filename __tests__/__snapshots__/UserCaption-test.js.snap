// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user caption snapshot test 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
        "textAlign": "left",
      },
      Array [
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
          "writingDirection": "ltr",
        },
        Array [
          Object {
            "fontSize": 16,
            "letterSpacing": 0.5,
            "lineHeight": 24,
            "marginVertical": 2,
          },
          undefined,
        ],
        Array [
          Object {
            "fontSize": 16,
            "letterSpacing": 0.5,
            "lineHeight": 24,
            "marginVertical": 2,
          },
          undefined,
        ],
      ],
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#3700b3",
        "fontWeight": "bold",
      }
    }
  >
    test
  </Text>
</Text>
`;
